### PR TITLE
Handle serialization of TryCast

### LINKED ISSUE #5691: `Expr::TryCast` not handled by logical plan serialization
### Is your feature request related to a problem or challenge?

Currently `Expr::TryCast` cannot be serialized to protobuf

### Describe the solution you'd like

Should be able to serialize it

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by thinkharderdev**
# Which issue does this PR close?

Closes #5691

# Rationale for this change

Handle serialization of `Expr::TryCast`

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/proto/src/logical_plan/mod.rs [hunk 1]
```diff
@@ -2386,6 +2386,15 @@ mod roundtrip_tests {
         roundtrip_expr_test(test_expr, ctx);
     }
 
+    #[test]
```
[the remark this answers is not in the document]
Comment (thinkharderdev) [reply datafusion/proto/src/logical_plan/mod.rs:2389]: you mean for `Expr::QualifiedWildcard`?

[the remark this answers is not in the document]
Comment (comphead) [reply datafusion/proto/src/logical_plan/mod.rs:2389]: sorry @thinkharderdev , I mean positive test case. 
ideally to have test case for valid cast and for invalid as currently in PR.

[the remark this answers is not in the document]
Comment (thinkharderdev) [reply datafusion/proto/src/logical_plan/mod.rs:2389]: not sure I follow. `Expr::Cast` was already handled. With this PR we should also be able to serialize `Expr::TryCast` so any valid expression should be serializable.

[the remark this answers is not in the document]
Comment (comphead) [reply datafusion/proto/src/logical_plan/mod.rs:2389]: Hey, yes, right, I meant to enclose 2 scenarios like
```
    fn roundtrip_try_cast() {
        -- invalid
        let test_expr =
            Expr::TryCast(TryCast::new(Box::new(lit(1.0_f32)), DataType::Boolean));

        let ctx = SessionContext::new();
        roundtrip_expr_test(test_expr, ctx);
        -- valid
        let test_expr =
            Expr::TryCast(TryCast::new(Box::new(lit(true)), DataType::Boolean));

        let ctx = SessionContext::new();
        roundtrip_expr_test(test_expr, ctx);
    }
```
Do you think it makes sense?

[the remark this answers is not in the document]
Comment (comphead) [reply datafusion/proto/src/logical_plan/mod.rs:2389]: oh wow, lol :) , in PG such query doesn't work 

```
❯ select 1::float::boolean;
+----------+
| Int64(1) |
+----------+
| true     |
+----------+
1 row in set. Query took 0.005 seconds.
```

PG: `cannot cast type double precision to boolean`

so then we probably can have a negative test to round trip try cast from string "blabla" to boolean?

[the remark this answers is not in the document]
Comment (thinkharderdev) [reply datafusion/proto/src/logical_plan/mod.rs:2389]: ah, I see what you mean now. I'm not sure that makes sense in this case though. This code is just for serialization/deserialization. So any `Expr` that can be constructed should be serialized/deserialized.

I added another case for an `Expr::TryCast` that would eval to `null` though

File: datafusion/proto/src/logical_plan/to_proto.rs [hunk 2]
```diff
@@ -489,50 +489,59 @@ impl TryFrom<&Expr> for protobuf::LogicalExprNode {
...
-                        .map(|ch| ch.to_string())
-                        .unwrap_or_default()
+                    escape_char: escape_char.map(|ch| ch.to_string()).unwrap_or_default(),
                 });
                 Self {
                     expr_type: Some(ExprType::Like(pb)),
                 }
             }
-            Expr::ILike(Like { negated, expr, pattern, escape_char }) => {
+            Expr::ILike(Like {
```
Comment (alamb) [inline datafusion/proto/src/logical_plan/to_proto.rs:518]: these changes look fine to me, but it isn't clear to me why `rustfmt` / lint didn't catch them before 🤔

File: datafusion/proto/src/logical_plan/to_proto.rs [hunk 3]
```diff
@@ -911,10 +943,12 @@ impl TryFrom<&Expr> for protobuf::LogicalExprNode {
                         data_type,
                     })),
                 }
-            },
+            }
 
-            Expr::QualifiedWildcard { .. } | Expr::TryCast { .. } =>
-                return Err(Error::General("Proto serialization error: Expr::QualifiedWildcard { .. } | Expr::TryCast { .. } not supported".to_string())),
+            Expr::QualifiedWildcard { .. } => return Err(Error::General(
```
Comment (alamb) [inline datafusion/proto/src/logical_plan/to_proto.rs:948]: I like the fact that `QualifiedWildcard` is the only expr left that isn't supported ❤️

**Review by alamb — APPROVED** [review]
Thank you @thinkharderdev

File: datafusion/proto/src/logical_plan/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (alamb) [reply datafusion/proto/src/logical_plan/mod.rs:2389]: Thank you for your diligence checking this PR @comphead 🏅

[the remark this answers is not in the document]
Comment (comphead) [reply datafusion/proto/src/logical_plan/mod.rs:2389]: Thanks @thinkharderdev

**Review by comphead — APPROVED** [review]
lgtm, thanks @thinkharderdev

**Pull request merged**